[vanguard] Add AddReportFab and change the description area of AddReportDrawer and EditReportDrawer to text area

peoplepeople000 opened the pull request:
# Description
Add AddReportFab and change the description area of AddReportDrawer and EditReportDrawer to text area

# Changes
Now there's an add report button on the map, and also the display of AddReportDrawer and EditReportDrawer look better

1. Add report button on the map
![截圖 2024-05-20 下午6 32 48](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/dcdab1cf-fc6c-4ac4-8bf0-102bd9ed8c74)

2. Add report drawer content
![截圖 2024-05-20 下午6 33 11](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/467713dc-1bd1-4ce7-8911-9876f39fdea5)

3. Edit report drawer content
![截圖 2024-05-20 下午6 33 28](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/cafc1685-561f-4db0-b055-e2396c5d27b7)

# Notes


# Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] Any changes to strings have been published to our translation tool
- [x] I conducted basic QA to assure all features are working
- [x] I requested code review from other team members

# Resolved Issues

peoplepeople000 committed:
feat: :lipstick: change the description row of EditReportDrawer to textarea

peoplepeople000 committed:
feat: :sparkles: Add a "ADD REPORT" fab on the map(haven't add any function yet)

peoplepeople000 committed:
feat: :lipstick: Change the look of addReportFab(it still can't work yet)

peoplepeople000 committed:
refactor: :fire: Remove clusterId in addReport() in src/store/report/index.ts and also remove the code that use clusterId in AddReportFab

peoplepeople000 committed:
fix: :bug: Fix some bugs about text display

I fixed the display bugs in AddReportDrawer

peoplepeople000 committed:
feat: :sparkles: Change the description area to text area

linyc0817 reviewed (changes requested):
Please provide some UI screenshot to the PR Readme. Let us check how will these features (Button, TextArea)  look like.

peoplepeople000 commented:
OK, I added some screenshots for the UI.

linyc0817 commented on src/components/Map/Fabs/AddReportFab/index.tsx:
```diff
@@ -0,0 +1,38 @@
+        handleAddReportClick();
+      }}
+    >
+      <img src={addReportBtn} alt="Add Report Button" />
```
According to your screenshot, img seems like didn't fit the button size. There is a gray border around it. Try to fix it.

peoplepeople000 replied:
OK, I had changed the UI. Bellow is how it looks now.
![截圖 2024-05-27 晚上7 02 55](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/4d502987-ee9c-4ee4-a190-6eb730d4282b)

linyc0817 commented on src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:
```diff
@@ -378,7 +378,8 @@ const StatusDescriptionAdd: React.FC = () => {
       </p>
-      <Input
+      <Textarea
+        minRows={1}
```
Don't let `textarea` only have one line, it just looks like `text input`.
I hope it can look like this. You can give it two or three rows as default to let user know what they have written when they input a lot.
 
![截圖 2024-05-21 上午2 14 39](https://github.com/CAMPUS-NYCU/smart-campus/assets/32840701/469485d7-a876-44ea-bebf-0328e42460c3)

peoplepeople000 replied:
OK, I had changed the minRows of add description field and edit description field. Also, I had modified the size of add photo button, in order to align with the text above. Bellow are how they look now.
![截圖 2024-05-27 晚上7 03 03](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/d4744d9c-416f-43cf-bcba-b0e1e1df0ac1)
![截圖 2024-05-27 晚上7 03 16](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/39d2cf1e-44b9-415d-a3f0-d32e8cb4dbaf)

linyc0817 reviewed (changes requested):
Some suggestions.
P.S. 上傳圖片的的黃色加號，讓它跟上下跟文字對齊（縮小它）

peoplepeople000 committed:
fix: :bug: Change the code of UI of add description, edit description,  add report button and add photo button

peoplepeople000 commented:
Bellow are the results of modified UI
1. Add report button
![image](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/9a8f18d2-c5cc-4ab4-b4f5-3839c091f637)
2. Add report drawer content
![image](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/9b559d80-fcfe-460e-abda-c49c46cb77ae)
3. Edit report drawer content
![image](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/308b153f-7678-44c3-ae0d-00b8f258afd2)

linyc0817 commented:
三個要改的部分
1. 我剛剛才發現不應該有多一個 `狀態敘述` (`回報敘述`)，這個版本要把原本是選項樣式的 `狀態敘述` 改成填空樣式。而不是新增一個新的欄位。
2. `狀態敘述` 的填空樣式，我覺得可以做成圖片這樣 (給底色，然後上半部記得對齊)
     <img width="200" alt="image" src="https://github.com/CAMPUS-NYCU/smart-campus/assets/32840701/15dd0988-8dc2-44b7-ad36-d52107c7d222">

3. `上傳圖片` 不是要整個 button 縮小，而是指裡面的那個黃色十字縮小而已(按鈕請回復到原本大小)

peoplepeople000 committed:
fix: :lipstick: Change some UI of AddReportDrawerContent and EditReportDrawerContent

1. Delete the row of "狀態敘述"   2.Change the style of description row.   3. Decrease the size of the plus sign before add photo

peoplepeople000 commented:
Bellow are the results of modified UI
1. Add report drawer content
![截圖 2024-06-12 下午5 41 05](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/657cf25f-c852-4194-bb1d-9bb7694248bd)

2. Edit report drawer content
![截圖 2024-06-12 下午5 41 24](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/b3d7dfb2-f5f6-4af5-9714-4e38644ebbbd)

linyc0817 reviewed (approved):
LGTM now，We can discuss how to modify the report interface in the future.
